sklego/model_selection.py

```diff
@@ -65,6 +68,20 @@ class TimeGapSplit:
 
         - `"rolling"` window has fixed size and is shifted entirely.
         - `"expanding"` left side of window is fixed, right border increases each fold.
+
+    Notes
+    -----
+    Native cross-dataframe support is achieved using
+    [Narwhals](https://narwhals-dev.github.io/narwhals/){:target="_blank"}.
+    Supported dataframes are:
+
+    - pandas
+    - Polars (eager)
+    - Modin
+
```

Comment:
Is CuDF breaking for some reason? It seems we are not doing any checking on X (`check_array` I am looking at you)

Reply:
again, thanks for asking - indeed, cuDF should be in the list, as `to_numpy()` is called explicitly here, and they do support that https://docs.rapids.ai/api/cudf/nightly/user_guide/api_docs/api/cudf.dataframe.to_numpy/

it's `__array__` (which gets triggered when you call `np.asarray` on an object) which they don't support they don't support https://github.com/koaning/scikit-lego/pull/665#discussion_r1597450090 . So, indeed, they should be included here